tests/MongoDB.Driver.Examples/InsertDataWithEncryptedFieldsWithAutoEncryption.cs

```diff
@@ -0,0 +1,295 @@
+                    })
+                },
+                {
+                    "insurance",
+                    new BsonDocument
+                    {
+                        { "policyNumber", 123142 },
+                        { "provider", "MaestCare" }
+                    }
+                }
+            };
+
+        // https://gist.github.com/DmitryLukyanov/8417d1f5c30f12ffeb2be29056aff455#file-insertdatawithencryptedfields-cs
+        [Fact]
+        public void InsertDataWithEncryptedFields()
+        {
+            var keyVaultNamespace = CollectionNamespace.FromFullName("encryption.__keyVault");
+            var medicalRecordsNamespace = CollectionNamespace.FromFullName("medicalRecords.patients");
+
+            var base64KeyId = ""; // paste the generated key in Standard GuidRepresentation form here
+
+            var ssnQuery = Builders<BsonDocument>.Filter.Eq("ssn", __sampleSsnValue);
+
+            // Construct a JSON Schema
+            var schema = JsonSchemaCreator.CreateJsonSchema(base64KeyId);
+
+            // Construct an encrypted client
+            var encryptedClient = CreateEncryptedClient(
+                medicalRecordsNamespace,
+                keyVaultNamespace,
+                schema);
+            var collection = encryptedClient
+                .GetDatabase(medicalRecordsNamespace.DatabaseNamespace.DatabaseName)
+                .GetCollection<BsonDocument>(medicalRecordsNamespace.CollectionName);
+
+            // Insert a document into the collection
+            collection.UpdateOne(ssnQuery, new BsonDocument("$set", SampleDoc), new UpdateOptions() { IsUpsert = true });
+            Console.WriteLine("Successfully upserted the sample document!");
+
+            // Query SSN field with encrypted client
+            var result = collection.Find(ssnQuery).Single();
+
+            Console.WriteLine("Encrypted client query by the SSN (deterministically-encrypted) field:\n" + result.ToJson());
+
+            // Query SSN field with normal client without encryption
+            var normalMongoClient = new MongoClient(__connectionString);
+            collection = normalMongoClient
+              .GetDatabase(medicalRecordsNamespace.DatabaseNamespace.DatabaseName)
+              .GetCollection<BsonDocument>(medicalRecordsNamespace.CollectionName);
+            var normalClientResult = collection.Find(ssnQuery).FirstOrDefault();
+            if (normalClientResult != null)
+            {
+                throw new Exception("Assert that the filtered data has not been found.");
+            }
+
+            // Query name (non-encrypted) field with normal client without encryption
+            var nameQuery = Builders<BsonDocument>.Filter.Eq("name", __sampleNameValue);
+            var normalClientNameResult = collection.Find(nameQuery).FirstOrDefault();
+            if (normalClientNameResult == null)
+            {
+                throw new Exception("Assert that the filtered data has been found.");
```

Comment:
I found the message confusing. Perhaps:

    throw new Exception("Expected the document to be found but none was found.");

Reply:
Not done?

                throw new Exception("Expected the document to be found but none was found.");